src/js/stores/MesosSummaryFetchers.tsx

```diff
@@ -0,0 +1,66 @@
+import * as React from "react";
+import { MESOS_SUMMARY_CHANGE } from "#SRC/js/constants/EventTypes";
+import MesosSummaryStore from "./MesosSummaryStore";
+import Node from "../structs/Node";
+
+const getNodeFromSummary = (nodeId?: string, summary?: any) => {
+  const trimmedNodeID = nodeId
```

Comment:
If you pass in node in stead of nodeId here, you can reuse getEncodedId and getId from there and drop all this trimming stuff